Panic when trying to migrate the git source to ArgoCD

I've the following GitRepository manifest file that doesn't include the secretRef field

```
apiVersion: source.toolkit.fluxcd.io/v1
kind: GitRepository
metadata:
  name: podinfo
  namespace: flux-system
spec:
  interval: 1m0s
  ref:
    branch: master
  url: https://github.com/stefanprodan/podinfo
```

However if I try to migrate this to Argo CD, I get the following error

```
>  mta kustomization --name podinfo                  
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x274a3f8]

goroutine 1 [running]:
github.com/akuity/mta/cmd.glob..func2(0xc0001d4a00?, {0x2c82acc?, 0x4?, 0x2c829dc?})
	/home/chernand/git/mta/cmd/kustomization.go:112 +0x538
github.com/spf13/cobra.(*Command).execute(0x40aa700, {0xc0002f9480, 0x2, 0x2})
	/home/chernand/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:944 +0x863
github.com/spf13/cobra.(*Command).ExecuteC(0x40aa9e0)
	/home/chernand/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(...)
	/home/chernand/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
github.com/akuity/mta/cmd.Execute()
	/home/chernand/git/mta/cmd/root.go:54 +0x1a
main.main()
	/home/chernand/git/mta/main.go:21 +0xf
```

--- a/.gitignore
+++ b/.gitignore
@@ -1,0 +1,2 @@
+.idea
+.vscode

--- a/cmd/kustomization.go
+++ b/cmd/kustomization.go
@@ -108,10 +108,18 @@ with kubectl.`,
 		}
 
 		//Get the secret holding the info we need
-		secret := &corev1.Secret{}
-		err = k.Get(ctx, types.NamespacedName{Namespace: kustomizationNamespace, Name: gitSource.Spec.SecretRef.Name}, secret)
-		if err != nil {
-			log.Fatal(err)
+		var sshPrivateKey string
+		if gitSource.Spec.SecretRef != nil && gitSource.Spec.SecretRef.Name != "" {
+			secret := &corev1.Secret{}
+			err = k.Get(ctx, types.NamespacedName{Namespace: kustomizationNamespace, Name: gitSource.Spec.SecretRef.Name}, secret)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			sshPrivateKey = string(secret.Data["identity"])
+		} else {
+			log.Info("Warning: SecretRef is not defined in the GitRepository spec. Proceeding without the SSH private key.")
+			sshPrivateKey = "" // Leave the SSHPrivateKey empty if the secret is not available
 		}
 
 		//	Argo CD ApplicationSet is sensitive about how you give it paths in the Git Dir generator. We need to figure some things out
@@ -145,7 +153,7 @@ with kubectl.`,
 			AppPath:                 "{{path}}",
 			AppDestinationServer:    "https://kubernetes.default.svc",
 			AppDestinationNamespace: kustomization.Spec.TargetNamespace,
-			SSHPrivateKey:           string(secret.Data["identity"]),
+			SSHPrivateKey:           sshPrivateKey,
 			GitOpsRepo:              gitSource.Spec.URL,
 		}
 

